tests/test_compr.cpp

```diff
@@ -0,0 +1,121 @@
+/**
+ * @file
+ *
+ * @author feliwir
+ *
+ * @brief Set of tests to validate the different compression algorithms
+ *
+ * @copyright Thyme is free software: you can redistribute it and/or
+ *            modify it under the terms of the GNU General Public License
+ *            as published by the Free Software Foundation, either version
+ *            2 of the License, or (at your option) any later version.
+ *            A full copy of the GNU General Public License can be found in
+ *            LICENSE
+ */
+#include <gtest/gtest.h>
+
+#include "always.h"
+#include "asciistring.h"
+#include "bufffileclass.h"
+#include "compressionmanager.h"
+#include "refpack.h"
+#include "zlibcompr.h"
+
+#define WRITE_DATA 0
+
+// Some special refpack mode that doesn't seem to be used by the game. Test it anyways :)
+TEST(compression, refpack_encode_quick)
+{
+    auto filepath = Utf8String(TESTDATA_PATH) + "/compr/uncompr.txt";
+    BufferedFileClass src_file(filepath.Str());
+    ASSERT_TRUE(src_file.Open(FM_READ));
+    size_t src_size = src_file.Size();
+    uint8_t *src_data = new uint8_t[src_size];
+    src_file.Read(src_data, src_size);
+
+    uint8_t *dst_data = new uint8_t[src_size + 8];
+    uint32_t dst_size = RefPack_Compress(dst_data, src_data, src_size, true);
+    EXPECT_GT(dst_size, 0);
+    EXPECT_LT(dst_size, src_size);
+
+    delete[] src_data;
```

Comment:
You can use `std::unique_ptr<>` to skip on `delete`. In several functions.

Reply:
Done